samples/snippets/cloud-client/src/main/java/privateca/UpdateCertificateTemplate.java

```diff
@@ -0,0 +1,116 @@
+
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+    // location: For a list of locations, see:
+    // https://cloud.google.com/certificate-authority-service/docs/locations
+    // certificateTemplateId: Id of the certificate template to update.
+    String project = "your-project-id";
+    String location = "ca-location";
+    String certificateTemplateId = "certificate-template-id";
+
+    updateCertificateTemplate(project, location, certificateTemplateId);
+  }
+
+  // Updates an existing certificate template.
+  public static void updateCertificateTemplate(
+      String project, String location, String certificateTemplateId)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    /* Initialize client that will be used to send requests. This client only needs to be created
+    once, and can be reused for multiple requests. After completing all of your requests, call
+    the `certificateAuthorityServiceClient.close()` method on the client to safely
+    clean up any remaining background resources. */
+    try (CertificateAuthorityServiceClient certificateAuthorityServiceClient =
+        CertificateAuthorityServiceClient.create()) {
+
+      String certificateTemplateName =
+          CertificateTemplateName.of(project, location, certificateTemplateId).toString();
+
+      // Set the parent name and the properties to be updated.
+      CertificateTemplate certificateTemplate =
+          CertificateTemplate.newBuilder()
+              .setName(certificateTemplateName)
+              .setIdentityConstraints(
+                  CertificateIdentityConstraints.newBuilder()
+                      .setAllowSubjectPassthrough(false)
+                      .setAllowSubjectAltNamesPassthrough(true)
+                      .build())
+              .build();
+
+      // Set the mask corresponding to the properties updated above.
+      FieldMask fieldMask =
+          FieldMask.newBuilder()
+              .addPaths("identity_constraints.allow_subject_alt_names_passthrough")
+              .addPaths("identity_constraints.allow_subject_passthrough")
+              .build();
+
+      /* Set the new template.
+      Set the mask to specify which properties of the template should be updated. */
+      UpdateCertificateTemplateRequest request =
+          UpdateCertificateTemplateRequest.newBuilder()
+              .setCertificateTemplate(certificateTemplate)
+              .setUpdateMask(fieldMask)
+              .build();
+
+      // Create the update certificate template request.
+      ApiFuture<Operation> futureCall =
+          certificateAuthorityServiceClient.updateCertificateTemplateCallable().futureCall(request);
+
+      Operation response = futureCall.get(60, TimeUnit.SECONDS);
+
+      // Check for errors.
+      if (response.hasError()) {
```

Comment:
I'm not sure what we should do here. We typically want to make sure that errors bubble up so that the developer can address them in some way. If this is a service-defined error, on the other hand, this might be the right thing to do.

Up to you: keep this as-is or throw a new exception with this message.

Reply:
If there is an exception in the operation, then that's thrown by futureCall.get().
Here, we are doing a passive service-based check. Hence, I would like to leave it as-is.